cpp/examples/basic_io.cpp

```diff
@@ -33,6 +33,9 @@ void check(bool condition)
   }
 }
 
+constexpr int NELEM = 1024;
```

Comment:
Should we add a comment explaining this value? In particular why 1024?

Reply:
This is just an arbitrary value, added a comment

Reply:
Ok wasn't sure if it was alignment related :)